src/lib/protocols/connections/ConnectionService.ts

```diff
@@ -96,58 +187,127 @@ class ConnectionService extends EventEmitter {
     const plainConnection = classToPlain(connection);
 
     const connectionResponse = new ConnectionResponseMessage({
-      threadId: message.id,
+      threadId: connectionRecord.tags.threadId!,
       connectionSig: await signData(plainConnection, this.wallet, connectionRecord.verkey),
     });
 
-    await this.updateState(connectionRecord, ConnectionState.RESPONDED);
+    await this.updateState(connectionRecord, ConnectionState.Responded);
     return createOutboundMessage(connectionRecord, connectionResponse);
   }
 
-  public async acceptResponse(messageContext: InboundMessageContext<ConnectionResponseMessage>) {
+  /**
+   * Process a received connection response message. This will not accept the connection request
+   * or send a connection acknowledgement message. It will only update the existing connection record
+   * with all the new information from the connection response message. Use {@link ConnectionService#createTrustPing}
+   * after calling this function to create a trust ping message.
+   *
+   * @param messageContext the message context containing a connetion response message
+   */
```

Comment:
I don't see `@returns` clause in docs for `processXxx` methods, is it on purpose?

Reply:
No not on purpose. This is because I first implemented all `processXXX` to not return anything but I later changed them all to return the updated connection record.